Jumpscale/clients/blockchain/tfchain/tests/200_reservation.py

```diff
@@ -0,0 +1,110 @@
+    c._explorer_get = explorer_client.explorer_get
+    c._explorer_post = explorer_client.explorer_post
+
+    # the devnet genesis seed is the seed of the wallet,
+    # which receives all block stakes and coins in the genesis block of the tfchain devnet
+    DEVNET_GENESIS_SEED = "image orchard airport business cost work mountain obscure flee alpha alert salmon damage engage trumpet route marble subway immune short tide young cycle attract"
+
+    # create a new devnet wallet
+    w = c.wallets.new("mywallet", seed=DEVNET_GENESIS_SEED)
+    # we create a new wallet using an existing seed,
+    # such that our seed is used and not a new randomly generated seed
+
+    # a tfchain (JS) wallet uses the underlying tfchain client for all its
+    # interaction with the tfchain network
+    assert w.network_type == "DEV"
+
+    w.capacity._notary_client_ = NotaryClientStub(c)
+
+    # create user threebot record
+    explorer_client.threebot_record_add(ThreeBotRecord(
+        identifier=1,
+        names=[BotName(value="user3bot")],
+        addresses=[],
+        public_key=w.key_pair_get(w.address).public_key,
+        expiration=1552581420,
+    ))
+    # create grid broker threebot record
+    explorer_client.threebot_record_add(ThreeBotRecord(
+        identifier=2,
+        names=[BotName(value="development.broker")],
+        addresses=[],
+        public_key=PublicKey.from_json("ed25519:e4f55bc46b5feb37c03a0faa2d624a9ee1d0deb5059aaa9625d8b4f60f29bcab"),
+        expiration=1552581420,
+    ))
+
+    broker_verify = nacl.signing.VerifyKey(
+        'e4f55bc46b5feb37c03a0faa2d624a9ee1d0deb5059aaa9625d8b4f60f29bcab',
+        encoder=nacl.encoding.HexEncoder)
+    broker_public = broker_verify.to_curve25519_public_key()
+
+    user_signing = w.capacity._threebot_singing_key('user3bot')
+    user_priv = user_signing.to_curve25519_private_key()
+    box = nacl.public.Box(user_priv, broker_public)
+
+    # try to reserve a 0-os VM
+    result = w.capacity.reserve_zos_vm("user@mail.com", "user3bot", 'ac1f6b47a04c')
+    assert result.submitted
+
+    reservation = w.capacity._notary_client.get(result.transaction.data.value.decode())
+    reservation = box.decrypt(reservation)
+    reservation = j.data.serializers.msgpack.loads(reservation)
+    schema = j.data.schema.get(url='tfchain.reservation.zos_vm')
+    o = schema.new(data=reservation)
+    assert o.type == 'vm'
+    assert o.size == 1
+    assert o.email == 'user@mail.com'
+    assert o.created <= j.data.time.epoch
```

Comment:
perhaps also assert some kind of lower bound?

Reply:
well since everything happens so fast, and the resolution is the second, I can't really. most of the time `o.created` is equal to the current epoch